packages/contracts/contracts/universal/MultiSigWallet.sol

```diff
@@ -0,0 +1,529 @@
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     * @param _owner Address resource requested.
+     */
+    modifier notConfirmed(uint _transactionId, address _owner) {
+        require(!confirmations[_transactionId][_owner]);
+        _;
+    }
+
+    /**
+     * @notice Ensure that the transaction not executed.
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     */
+    modifier notExecuted(uint _transactionId) {
+        require(!transactions[_transactionId].executed);
+        _;
+    }
+
+    /**
+     * @notice Ensure that the address is not zero address.
+     *
+     * @param _address Address resource requested.
+     */
+    modifier notNull(address _address) {
+        require(_address != address(0));
+        _;
+    }
+
+    /**
+     * @notice Ensure that the number of confirmations required is valid.
+     *
+     * @param _ownerCount A number of owners.
+     * @param _numConfirmationsRequired Number of required confirmations.
+     */
+    modifier validNumConfirmations(uint _ownerCount, uint _numConfirmationsRequired) {
+        require(
+            _numConfirmationsRequired <= _ownerCount
+            && _numConfirmationsRequired != 0
+            && _ownerCount != 0);
+        _;
+    }
+
+    /**
+    * @notice Initializer.
+    *
+    * @param _owners List of initial owners.
+    * @param _numConfirmationsRequired Number of required confirmations.
+    */
+    function initialize(address[] memory _owners, uint _numConfirmationsRequired) public virtual onlyInitializing {
+        require(_owners.length > 0, "MultiSigWallet: owners requred");
+        for (uint i = 0; i < _owners.length; i++) {
+            require(!isOwner[_owners[i]], "MultiSigWallet: owner already exists");
+            require(_owners[i] != address(0), "MultiSigWallet: invalid owner address");
+            isOwner[_owners[i]] = true;
+        }
+        owners = _owners;
+        numConfirmationsRequired = _numConfirmationsRequired;
+    }
+
+    /**
+    * @notice Allows to add a new owner. Transaction has to be sent by wallet.
+    *
+    * @param _owner Address of new owner.
+    */
+    function addOwner(address _owner)
+        external
+        override
+        onlyWallet
+        ownerDoesNotExist(_owner)
+        notNull(_owner)
+        validNumConfirmations(owners.length + 1, numConfirmationsRequired)
+    {
+        isOwner[_owner] = true;
+        owners.push(_owner);
+        emit OwnerAddition(_owner);
+    }
+
+    /**
+    * @notice Allows to remove an owner. Transaction has to be sent by wallet.
+    *
+    * @param _owner Address of owner.
+    */
+    function removeOwner(address _owner)
+        external
+        override
+        onlyWallet
+        ownerExists(_owner)
+    {
+        isOwner[_owner] = false;
+        uint _index;
+        uint i;
+        /// find array index by item
+        for (i = 0; i < owners.length - 1; i++) {
```

Comment:
Please check other for loop example.

Reply:
I updated it along with some other code in a similar vein.